dft/src/butterflies.rs

```diff
@@ -3,14 +3,50 @@ use core::mem::MaybeUninit;
 use itertools::izip;
 use p3_field::{Field, PackedField, PackedValue};
 
+/// A butterfly operation used in NTT to combine two values into a new pair.
+///
+/// This trait defines how to transform two elements (or vectors of elements)
+/// according to the structure of a butterfly gate.
+///
+/// In an NTT, butterflies are the core units that recursively combine values
+/// across layers. Each butterfly computes:
+///
+///   (a + b * twiddle, a - b * twiddle)   // DIT
+/// or
+///   (a + b, (a - b) * twiddle)           // DIF
+///
+/// The transformation can be applied:
+/// - in-place (mutating input values)
+/// - to full rows of values (arrays of field elements)
+/// - out-of-place (writing results to separate destination buffers)
+///
+/// Different butterfly variants (DIT, DIF, or twiddle-free) define the exact formula.
 pub trait Butterfly<F: Field>: Copy + Send + Sync {
+    /// Applies the butterfly transformation to two packed field values.
+    ///
+    /// This method takes two inputs `x_1` and `x_2` and returns two outputs `(y_1, y_2)`
+    /// depending on the butterfly type.
+    ///
+    /// Example (DIF):
+    ///   Input:  x_1 = a, x_2 = b
+    ///   Output: (a + b, (a - b) * twiddle)
```

Comment:
```suggestion
    /// ```text
    /// Example (DIF):
    ///   Input:  x_1 = a, x_2 = b
    ///   Output: (a + b, (a - b) * twiddle)
    /// ```
```